src/redux/Missions/missionsSlice.js

```diff
@@ -0,0 +1,74 @@
+import axios from "axios"; // eslint-disable-line
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const url = 'https://api.spacexdata.com/v3/missions';
+const initialState = {
+  missions: [],
+  status: 'idle',
+  error: null,
+};
+
+/* eslint-disable */
+const missionsSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    missionAdded(state, action) {
+      state.missions = action.payload;
+    },
+    reserveMission(state, action) {
+      const id = action.payload;
+      const newArr = state.missions.map((item) => {
+        if (item.mission_id === id) {
+          item.reserved = true;
+        }
+        return item;
+      });
+      console.log(newArr);
+      state.missions = newArr;
```

Comment:
I think it's best to not mutate state directly and use **return** as they mention in #30 .
We know that even if we mutated state directly **[immer](https://redux-toolkit.js.org/usage/immer-reducers#why-immer-is-built-in)** will handle automatically but our **microverse reviewer** will still going follow the requirement mention in #30 :smile: .